packages/loader/container-loader/src/container.ts

```diff
@@ -1096,7 +1139,9 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
 
         // pre-0.58 error message: existingContextDoesNotSatisfyIncomingProposal
-        this.close(new GenericError("Existing context does not satisfy incoming proposal"));
+        const error = new GenericError("Existing context does not satisfy incoming proposal");
+        this.close(error);
```

Comment:
There are still some calls to this.close in this class that aren't followed by this.dispose.  Is this intentional?

Also, it highlights the point that the behavior of close _does_ change slightly here, as you called out... around cleaning up event handlers.  not sure what to make of that.

Reply:
I guess that's kind of unavoidable, since Container also has the "disposed" event.  Seems fine.

Reply:
I left a suggestion in DM about having DM.close(error, false) turn around and invoke the Dispose codepath as well.  This would really pinpoint the semantics here - _for now_ close doesn't change (it also disposes everything), but later that will change.  And if you do that, then I _think_ that anyone calling Container.close will get truly unchanged behavior - well, they'll just get the additional dispose event.  This feels correct to me, what do you think?